goldstone/core/models.py

```diff
@@ -882,31 +801,34 @@ def clouddata():
         # Availability Zone for each one.
         result = []
 
+        # For every found host...
         for host in hosts:
             parsed_name = Host._parse_host_name(host["host_name"])[0]
 
             if all(x["host_name"] != parsed_name for x in result):
-                # This is a new entry for the result set.
+                # This is a new entry for the result set. Set the host_name
+                # value, and concoct a unique id for it.
                 host["host_name"] = parsed_name
+                host[cls.unique_cloud_id_key] = cls.unique_cloud_id(host)
+
+                # We don't want or need this key.
                 del host["service"]
```

Comment:
> IIRC, the openstack model is a bit flaky here, in that it will give you multiple host entries with the same name, but a different service.

Yes. This method filters out multiple host entries in the data returned from nova_client.hosts.list(), for exactly this reason.

> Let me if host --> nova service is missing from the model and I'll ticket it separately.

I think your assertion is correct.  Whether or not it's ticketed now, we would bump into this when we start using and debugging these API endpoints for the first time.